### PR TITLE
Problem: cometbft not up to date

### PR DESCRIPTION
**Pull request by yihuang**
update to 0.38.13

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **Documentation**
	- Updated `CHANGELOG.md` to include a new section titled "UNRELEASED" for tracking improvements.

- **Chores**
	- Updated various dependencies to their latest versions for improved compatibility and performance.
	- Updated Go version requirements to ensure alignment with the latest toolchain.
	- Updated `flake.nix` to reference a stable release of `nixpkgs`.
	- Updated `nix/sources.json` to reflect changes in the `nixpkgs` package source.
	- Increased timeout settings for linting processes to allow for longer checks.
	- Added `golangci-lint` package to the build overlay for enhanced linting capabilities.
	- Introduced a new Nix expression for the `golangci-lint` tool, facilitating easier linting integration.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## UNRELEASED
+
+### Improvements
+
+* [#]() Update cometbft to 0.38.13.
```
Comment (yihuang) [inline CHANGELOG.md:7]:
```suggestion
* [#1664](https://github.com/crypto-org-chain/cronos/pull/1664) Update cometbft to 0.38.13.
```

**Review by mmsqe — APPROVED** [review]
https://github.com/cometbft/cometbft/compare/fdf90d16f608...v0.38.13
https://github.com/cometbft/cometbft-db/compare/v0.12.0...v0.14.1

File: go.mod [hunk 2]
```diff
@@ -1,8 +1,8 @@
 module github.com/crypto-org-chain/cronos/v2
 
-go 1.22.2
+go 1.22.7
```
Comment (yihuang) [inline go.mod:3]:
```suggestion
go 1.22.2
```

File: go.mod [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
 
-toolchain go1.22.3
+toolchain go1.22.8
```
Comment (yihuang) [inline go.mod:5]:
```suggestion
toolchain go1.22.3
```

**Pull request merged**